classes/robot/crawler.php

```diff
@@ -286,6 +286,10 @@ public function mark_for_crawl($baseurl, $url, $courseid = null, $priority = TOO
 
         $url = $this->absolute_url($baseurl, $url);
 
+        if (filter_var($url, FILTER_VALIDATE_URL) === false) {
```

Comment:
This should probably use the moodle libs instead:

clean_param($url, PARAM_URL);